callstack/react-native-builder-bob
# refactor: add separate option to generate view modules

### Summary

After merging [this feature](https://github.com/react-native-community/bob/pull/64), `bob` includes both Functional and UI native modules upon bootstraping. Not all projects require both types though, so additional command to select all/one/none would be great to have.

- Added a separate template folders (`native-view-library`, `objc-view-library`,  `swift-view-library`) to add only necessary files to the generated project 
- Added new type `ModuleType` to distinguish, what type of module dev would like to generate 
- The mixed case (view module with native module) has not been added here, because in most cases ppl create only a separate modules (We can consider to implement it in separate PR)

Closes #93 

### Test plan
- run below command for all types 

```sh
../bob/bin/bob create test-project
```

- run `yarn bootstrap` and then `yarn android` & `yarn ios`
- check, if all projects has been generated correctly 

### Screenshots 

Native View Module 
![Screen Shot 2020-11-13 at 1 05 29 PM](https://user-images.githubusercontent.com/12766071/99079842-c5f44280-25c0-11eb-90f8-9b3c48e8e6ed.png)

Native module
![Screen Shot 2020-11-13 at 12 51 01 PM](https://user-images.githubusercontent.com/12766071/99079909-d86e7c00-25c0-11eb-9709-e8ef908f0593.png)

Added below two options:
![Screen Shot 2020-11-13 at 3 00 33 PM](https://user-images.githubusercontent.com/12766071/99080080-110e5580-25c1-11eb-8eb6-2c884563b5f9.png)
